### DISCUSSION TITLE
How to set the max columns to more than 20?

### DISCUSSION THREAD

**Discussion by nvamsimohan** [approximatelabs/sketch#23] [Q&A] [answered]
Hi!, I'm using the latest version of sketch and when I tried to process a dataset with more than 20 columns, I got an error - ValueError: Too many columns (57), max is 20 in current version (set SKETCH_MAX_COLUMNS to override)

Where can I set this SKETCH_MAX_COLUMNS  parameter please?
I tried
import os
os.environ['SKETCH_MAX_COLUMNS'] = 32

But it is not working.

Any help is greatly appreciated.

Thanks much!

**Comment by bluecoconut** [comment, marked as answer]
Hi @nvamsimohan , what you did should work. 

It might be the case you have to pass it as a string? eg. `os.environ['SKETCH_MAX_COLUMNS'] = '32'`

This is used by https://github.com/approximatelabs/sketch/blob/89348454851a0902f26fdabe3a9ec55407b2ee46/sketch/pandas_extension.py#L147

**Reply by nvamsimohan** [reply]
Thank you! that worked. It seems for > 20 columns, we have to use our own API key.
Now, I see this message - "Consider setting SKETCH_USE_REMOTE_LAMBDAPROMPT=False 
and run with your own open-ai key"
